test/Unit/Queue/TimeLimitedQueueTest.php

```diff
@@ -0,0 +1,113 @@
+        $queue->store($this->newMessage('Test 1', 1024));
+        $queue->store($this->newMessage('Test 2', 2048));
+        $queue->store($this->newMessage('Test 3', 3072));
+        $queue = new TimeLimitedQueue($queue, 0);
+
+        $transport = $this->createMock(TransportInterface::class);
+        $transport
+            ->expects($this->at(0))
+            ->method('send')
+            ->with($this->callback(function (GenericMessage $message) {
+                $this->assertEquals('Test 1', $message->getHeader('subject')[0]->getValue());
+                return true;
+            }));
+
+        $transport
+            ->expects($this->at(1))
+            ->method('send')
+            ->with($this->callback(function (GenericMessage $message) {
+                $this->assertEquals('Test 2', $message->getHeader('subject')[0]->getValue());
+                return true;
+            }));
+
+        $transport
+            ->expects($this->at(2))
+            ->method('send')
+            ->with($this->callback(function (GenericMessage $message) {
+                $this->assertEquals('Test 3', $message->getHeader('subject')[0]->getValue());
+                return true;
+            }));
+
+        $processor = new QueueProcessor($transport, [$queue]);
+        $count = $processor->process();
+
+        $this->assertCount(0, $storage);
+        $this->assertSame(3, $count);
+    }
+
+    /**
+     * @test
+     */
+    public function it_will_expire_messages_when_time_limit_is_reached(): void
+    {
+        $storage = new \ArrayObject();
+
+        $queue = new ArrayObjectQueue($storage);
+        $queue->store($this->newMessage('Test 1', 1024));
+        $queue->store($this->newMessage('Test 2', 2048));
+        $queue->store($this->newMessage('Test 3', 3072));
+        $queue = new TimeLimitedQueue($queue, 2000);
+
+        $transport = $this->createMock(TransportInterface::class);
+        $transport
+            ->expects($this->once())
+            ->method('send')
+            ->with($this->callback(function (GenericMessage $message) {
+                $this->assertEquals('Test 1', $message->getHeader('subject')[0]->getValue());
+                return true;
+            }));
+
+        $processor = new QueueProcessor($transport, [$queue]);
+        $count = $processor->process();
+
+        $this->assertCount(0, $storage);
+        $this->assertSame(1, $count);
+    }
+
+    protected function getDecoratingQueue(QueueInterface $queue): QueueInterface
+    {
+        return new TimeLimitedQueue($queue, 0);
+    }
+
+    /**
+     * @param string $subject
+     * @param int|null $age
+     * @return MessageInterface
+     */
+    private function newMessage(string $subject, int $age = 0): MessageInterface
+    {
+        return (new GenericMessage())
+            ->withHeader(new GenericHeader(
+                QueueIfFailedTransport::QUEUED_HEADER,
+                (new DateTime('now'))->modify("-$age seconds")->format('r')
```

Comment:
Sorry, I rather use immutable objects, everywhere. I know this is just a test, and it might be silly, but I believe signalling helps.

Reply:
`<insert facepalm emoji>` Yep, and I know that too … early morning typo.